applications/StructuralMechanicsApplication/custom_constitutive/elastic_isotropic_3d.cpp

```diff
@@ -389,4 +396,51 @@ void ElasticIsotropic3D::CalculateCauchyGreenStrain(
     noalias(rStrainVector) = MathUtils<double>::StrainTensorToVector(E_tensor);
 }
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+void ElasticIsotropic3D::AddThermalEffect(
+    ConstitutiveLaw::Parameters& rValues,
+    Vector& rStrainVector
+    )
+{
+    const auto &r_properties = rValues.GetMaterialProperties();
+    const auto &r_geometry = rValues.GetElementGeometry();
+
+    if (r_properties.Has(THERMAL_EXPANSION_COEFFICIENT)) {
+        const double alpha = r_properties[THERMAL_EXPANSION_COEFFICIENT];
+        if (r_geometry[0].SolutionStepsDataHas(TEMPERATURE)) {
+            const double gauss_point_temperature = ConstitutiveLawUtilities<VoigtSize>::
+                GetNodalVariableOnIntegrationPoint(TEMPERATURE, rValues);
+            const double gauss_point_ref_temperature = ConstitutiveLawUtilities<VoigtSize>::
+                GetNodalVariableOnIntegrationPoint(REFERENCE_TEMPERATURE, rValues);
```

Comment:
I would say `REFERENCE_TEMPERATURE` should be in the `ProcessInfo` Or did I misunderstand sth?

Reply:
i woukd keep it in the oroperties: imagine that you want to merge teo material which are initially at two different temperatures (you woukd want the strain to be zero at the moment at which you put them into contact)